libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F4/serial_api.c

```diff
@@ -479,4 +631,537 @@ void serial_break_clear(serial_t *obj)
+* @return internal NVIC RX DMA IRQ index of U(S)ART peripheral
+*/
+IRQn_Type h_serial_rx_get_irqdma_index(serial_t *obj)
+{
+    IRQn_Type irq_n = (IRQn_Type)0;
+
+    UartHandle.Instance = (USART_TypeDef *)SERIAL_OBJ(uart);
+
+    switch (SERIAL_OBJ(uart)) {
+#if defined(USART1_BASE)
+        case UART_1:
+            irq_n = DMA2_Stream5_IRQn;
+            break;
+#endif
+#if defined(USART2_BASE)
+        case UART_2:
+            irq_n = DMA1_Stream5_IRQn;
+            break;
+#endif
+#if defined(USART3_BASE)
+        case UART_3:
+            irq_n = DMA1_Stream1_IRQn;
+            break;
+#endif
+#if defined(UART4_BASE)
+        case UART_4:
+            irq_n = DMA1_Stream2_IRQn;
+            break;
+#endif
+#if defined(UART5_BASE)
+        case UART_5:
+            irq_n = DMA1_Stream0_IRQn;
+            break;
+#endif
+#if defined(USART6_BASE)
+        case UART_6:
+            irq_n = DMA2_Stream2_IRQn;
+            break;
+#endif
+#if defined(UART7_BASE)
+        case UART_7:
+            irq_n = DMA1_Stream3_IRQn;
+            break;
+#endif
+#if defined(UART8_BASE)
+        case UART_8:
+            irq_n = DMA1_Stream6_IRQn;
+            break;
+#endif
+        default:
+            irq_n = (IRQn_Type)0;
+    }
+
+    return irq_n;
+}
+#endif
+//----------------------------------------------------------------------------------------
+// MBED API FUNCTIONS
+//----------------------------------------------------------------------------------------
+
+/** Begin asynchronous TX transfer. The used buffer is specified in the serial object,
+ *  tx_buff
+ *
+ * @param obj       The serial object
+ * @param tx        The buffer for sending
+ * @param tx_length The number of words to transmit
+ * @param tx_width  The bit width of buffer word
+ * @param handler   The serial handler
+ * @param event     The logical OR of events to be registered
+ * @param hint      A suggestion for how to use DMA with this transfer
+ * @return Returns number of data transfered, or 0 otherwise
+ */
+int serial_tx_asynch(serial_t *obj, const void *tx, size_t tx_length, uint8_t tx_width, uint32_t handler, uint32_t event, DMAUsage hint)
+{
+    
+    uint32_t tmpstatus = 0;
+  
+    // Check buffer is ok
+    MBED_ASSERT(tx != (void*)0);
+    MBED_ASSERT(tx_width == 8); // support only 8b width
+
+    if (tx_length == 0) return 0;
+  
+    // Set up buffer
+    h_serial_tx_buffer_set(obj, (void *)tx, tx_length, tx_width);
+  
+    // Set up events
+    h_serial_tx_enable_event(obj, SERIAL_EVENT_TX_ALL, 0); // Clear all events
+    h_serial_tx_enable_event(obj, event, 1); // Set only the wanted events
+  
+    // Enable interrupt
+    IRQn_Type irqn = h_serial_get_irq_index(obj);
+    NVIC_ClearPendingIRQ(irqn);
+    NVIC_DisableIRQ(irqn);
+    NVIC_SetPriority(irqn, 1);
+    NVIC_SetVector(irqn, (uint32_t)handler);
+    UartHandle.Instance = (USART_TypeDef *)SERIAL_OBJ(uart);
+    NVIC_EnableIRQ(irqn);
+
+#if DEVICE_SERIAL_ASYNCH_DMA
```

Comment:
I'd rather like to keep it. 
In case you have a DMA conflict (if an application wants to use DMA for anything else), you just need to set the define to 0 and use normal interrupt mode, withou DMA.